src/services/codefixes/convertTypedefToType.ts

```diff
@@ -0,0 +1,253 @@
+import {
+    Diagnostics,
+    factory,
+    flatMap,
+    getTokenAtPosition,
+    HasJSDoc,
+    hasJSDocNodes,
+    InterfaceDeclaration,
+    isInJSDoc,
+    isJSDocTypedefTag,
+    JSDocPropertyTag,
+    JSDocTypedefTag,
+    JSDocTypeExpression,
+    JSDocTypeLiteral,
+    mapDefined,
+    Node,
+    ParenthesizedTypeNode,
+    PropertySignature,
+    some,
+    SourceFile,
+    SyntaxKind,
+    textChanges,
+    TypeAliasDeclaration,
+    TypeNode,
+    TypeNodeSyntaxKind,
+    UnionTypeNode,
+} from "../_namespaces/ts";
+import { codeFixAll, createCodeFixAction, registerCodeFix } from "../_namespaces/ts.codefix";
+
+const fixId = "convertTypedefToType";
+const errorCodes = [Diagnostics.JSDoc_typedef_may_be_converted_to_TypeScript_type.code];
+registerCodeFix({
+    fixIds: [fixId],
+    errorCodes,
+    getCodeActions(context) {
+        const node = getTokenAtPosition(
+            context.sourceFile,
+            context.span.start
+        );
+        if (!node) return;
+        const changes = textChanges.ChangeTracker.with(context, t => doChange(t, node, context.sourceFile));
+
+        if (changes.length > 0) {
+            return [
+                createCodeFixAction(
+                    fixId,
+                    changes,
+                    Diagnostics.JSDoc_typedef_may_be_converted_to_TypeScript_type,
+                    fixId,
+                    Diagnostics.JSDoc_typedefs_may_be_converted_to_TypeScript_types

```

Comment:
These are the texts that show up in the quick fix menu and should use a different pair of diagnostics than the suggestion, phrased as an imperative (“Convert typedef to TypeScript type”).

See the `require` to `import` fix for example:

<img width="582" alt="image" src="https://user-images.githubusercontent.com/3277153/212976196-41559cd3-bad9-4368-a612-4fc670d9c407.png">


<img width="367" alt="image" src="https://user-images.githubusercontent.com/3277153/212976239-327e3e5f-cc5f-44ac-9e19-2ae94af8752a.png">

Reply:
changed